logstreams/src/main/java/io/zeebe/logstreams/storage/atomix/AtomixLogStorageReader.java

```diff
@@ -50,11 +49,13 @@ public boolean isEmpty() {
 
   @Override
   public long read(final DirectBuffer readBuffer, final long address) {
-    if (address < reader.getFirstIndex()) {
+    final long index = reader.reset(address);
```

Comment:
This reset can also be avoided by checking `reader.getNextEntry`

Reply:
And that was the big one? That's...unexpected :sweat_smile: